tasks/kernel_matrix_testing/compiler.py

```diff
@@ -30,8 +35,8 @@ def start_compiler(ctx):
         "docker run -d --restart always --name kmt-compiler --mount type=bind,source=./,target=/datadog-agent kmt:compile sleep \"infinity\""
     )
 
-    uid = ctx.run("id -u").stdout.rstrip()
-    gid = ctx.run("id -g").stdout.rstrip()
+    uid = cast(Result, ctx.run("id -u")).stdout.rstrip()
```

Comment:
Will this cause a runtime failure if the command fails?

Reply:
The cast doesn't, it's a noop in runtime, and `run` should always return a value even if the comand fails, as we're not using detached commands.

Reply:
So what is the point of the cast here?

Reply:
For shortness, instead of doing

```
res = ctx.run(...)
if res is None:
    raise RuntimeError()
uid = res.stdout.strip()
``` 

we just have the cast to force the typing system to ignore that `ctx.run` could return `None` (even though it will never return it with these arguments).